flink-core/src/main/java/org/apache/flink/configuration/JMXServerOptions.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.annotation.docs.Documentation;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * The set of configuration options relating to heartbeat manager settings.
+ */
+@PublicEvolving
+public class JMXServerOptions {
+
+	/** Port configured to enable JMX server for metrics and debugging. */
+	@Documentation.Section(Documentation.Sections.EXPERT_DEBUGGING_AND_TUNING)
+	public static final ConfigOption<String>JMX_SERVER_PORT =
+		key("jmx.server.port")
+			.defaultValue("-1")
```

Comment:
having no default value at all would make this easier to understand imo

Reply:
hmm this is the part I am also struggling. --> how to make the original `JMXReporter.port` configuration work as a "deprecated" but still working alternative. 

ideally speaking this config should override the ones in `JMXReporter`, but the config string for that is actually dynamic. e.g. `jmxreporter.<my_jmxrepoter_name>.port` --> makes it hard to fallback. 

I was trying to use a "-1" here to indicate that a global JMXServer singelton should not be initialized during the taskmanager / jobmaster startup, instead differ to the JMXReporter startup. (that's where the "<my_reporter_name>" config can be accessed.

Reply:
But the current code handles this fine though, right?

You just have to replace
```
portsConfig.equals(JMXServerOptions.JMX_SERVER_PORT.defaultValue())
```
with
```
portsConfig == null
```
in `JMXServer#startInstance(string)`. That's ultimately all I was suggesting.